core/services/pipeline/common.go

```diff
@@ -242,20 +241,11 @@ func (js JSONSerializable) MarshalJSON() ([]byte, error) {
 	}
 	switch x := js.Val.(type) {
 	case []byte:
-		// Don't need to HEX encode if it is a valid JSON string
 		if json.Valid(x) {
 			return json.Marshal(string(x))
 		}
-
-		// Don't need to HEX encode if it is already HEX encoded value
-		if utils.IsHexBytes(x) {
-			return json.Marshal(string(x))
-		}
-
-		return json.Marshal(hex.EncodeToString(x))
```

Comment:
This looks suspicious to me, we encoded to hex here, but did not decode anywhere else.
Maybe I am missing an important case here, I rely on @begmaroman input here, who wrote this code.